### PR TITLE
Josh/cf 27 create data model for request event

### PR DESCRIPTION
**Pull request by JoshuaWilkes**
image

Work in progress, testing this approach for audit logs on requests.
RCF

### CODE REVIEW

File: pkg/eventhandler/eventhandler.go [hunk 1]
```diff
@@ -50,28 +51,36 @@ func (n *EventHandler) HandleGrantEvent(ctx context.Context, log *zap.SugaredLog
...
+	if oldStatus == newStatus {
+		// I anticipate that this would be succeptible to a race condition, recoverable if the eventbridge retries the event handler
+		// this is because the grant events are sourced from the access handler prior to the request being saved to dynamodb on creation
+		// we could solve this by saving the request to the DB prior to making the call to the access handler?
+		if event.DetailType == gevent.GrantCreatedType {
+			requestEvent := access.NewGrantCreatedEvent(gq.Result.ID, time.Now(), nil)
+			log.Infow("inserting request event for grant created")
+			return n.db.Put(ctx, &requestEvent)
+		}
+		return nil
```
Comment (chrnorm) [inline pkg/eventhandler/eventhandler.go]: I found this a bit confusing to read - we could make it simpler by reducing the levels of nesting

ie

```go
if oldStatus != newStatus {
  // emit status change event
}

if event.DetailType == gevent.GrantCreatedType {
  // emit grant created event
}
```

**Comment by chrnorm** [comment]
Agreed on the approach and the API design here 👍

**Comment by JoshuaWilkes** [comment]
With all events 
image

**Comment by chrnorm** [comment]
Looks good - a UI nitpick on this, can you use lowercase instead of all caps for the status? e.g. "pending" rather than "PENDING"

**Review by ellie-narducci — APPROVED** [review]

**Pull request merged**